src/main/java/com/clearspring/analytics/stream/cardinality/RegisterSet.java

```diff
@@ -22,23 +22,36 @@
     public final static int REGISTER_SIZE = 5;
 
     public final int count;
-    public final int size;
 
-    private final int[] M;
+    private final RegisterSetStorage M;
 
     public RegisterSet(int count) {
         this(count, null);
     }
 
     public RegisterSet(int count, int[] initialValues) {
+        this(count, initialValues, RegisterSetStorage.Type.ARRAY_BACKED);
+    }
+
+    public RegisterSet(int count, int[] initialValues, RegisterSetStorage.Type type) {
         this.count = count;
 
-        if (initialValues == null) {
-            this.M = new int[getSizeForCount(count)];
-        } else {
-            this.M = initialValues;
+        switch (type) {
+            case ARRAY_BACKED:
+                if (initialValues == null)
```

Comment:
more braces. also, what's with the pointless exhaustive case analysis? you've got at least two layers of indirection here. Even the most obvious thing would be to give the enums a creator method,